src/com/xilinx/rapidwright/design/NetTools.java

```diff
@@ -144,4 +144,18 @@ public static List<NodeTree> getNodeTrees(Net net) {
 
         return subtrees;
     }
+    
+    /**
+     * Checks if the provided net drives a clock site pin input.
+     * 
+     * @param net The net to examine.
+     * @return True if the net has a site pin clock input, false otherwise.
+     */
+    public static boolean hasClockSinks(Net net) {
+        for (SitePinInst sink : net.getPins()) { 
+            if (sink.isOutPin()) continue;
+            if (sink.getName().contains("CLK")) return true;
```

Comment:
Would this work?
```suggestion
            if (sink.getName().startsWith("CLK")) return true;
```

Reply:
Here are all the site pin inputs across all sitetypes/devices/series that contain "CLK":
[all_site_pins_contain_clk.txt](https://github.com/user-attachments/files/17984849/all_site_pins_contain_clk.txt)

I think we should probably stick with `.contains()`.